pubsub.go

```diff
@@ -268,3 +342,49 @@ func bootstrapPubsub(ctx context.Context, cr routing.ContentRouting, host p2phos
 
 	wg.Wait()
 }
+
+func (wg *watchGroup) add(ctx context.Context, p *PubsubValueStore, key string, outCh chan []byte) {
+	proxy := make(chan []byte, 1)
+
+	go func() {
+		ctx, cancel := context.WithCancel(ctx)
+		wg.listeners[outCh] = ctx
```

Comment:
I think this needs to happen *outside* the goroutine (while we're holding the lock).